assembly/std/index.ts

```diff
@@ -102,6 +102,94 @@ export function balanceOf(address: string): u64 {
   return env.balanceOf(address);
 }
 
+/**
+ * Check for key in datastore
+ *
+ * @param {StaticArray<u8>} key
+ *
+ * @return {bool} - true if key is present in datastore, false otherwise.
+ */
+export function hasOpKey(key: StaticArray<u8>): bool {
+  let result = env.hasOpKey(key);
+  // From https://doc.rust-lang.org/reference/types/boolean.html &&
+  // https://www.assemblyscript.org/types.html
+  // we can safely cast from u8 to bool
+  return bool(result[0]);
+}
+
+/*
+ * Get data associated with the given key from datastore
+ *
+ * @param {StaticArray<u8>} key
+ *
+ * @return {StaticArray<u8>} - data as a byte array
+ */
+export function getOpData(key: StaticArray<u8>): StaticArray<u8> {
+  return env.getOpData(key);
+}
+
+/*
+ * Get all keys from datastore
+ *
+ * @return {Array<StaticArray<u8>} - a list of key (e.g. a list of bytearray)
+ */
+export function getOpKeys(): Array<StaticArray<u8>> {
+  let keys_ser = env.getOpKeys();
+  return derOpKeys(keys_ser);
+}
+
+export function derOpKeys(keys_ser: StaticArray<u8>): Array<StaticArray<u8>> {
+
+  let MAX_DATASTORE_ENTRY_COUNT = u16.MAX_VALUE;
+  let default_res = new Array<StaticArray<u8>>();
+  let keys_der = new Array<StaticArray<u8>>();
+  // Datastore deserialization
+  // Format is: L (u32); V1_L (u8); V1 data (u8*V1_L); ...
+  // TODO: test keys_ser initial len
+  // u8 * 4 (LE) => u32
+  let entry_count: u32 = keys_ser[0] << 24 + keys_ser[1] << 16 + keys_ser[2] << 8 + keys_ser[3];
+
+  if (entry_count == 0 || entry_count > MAX_DATASTORE_ENTRY_COUNT) {
```

Comment:
in the case `entry_count > MAX_DATASTORE_ENTRY_COUNT` for now we should `throw` (to avoid silent unexpected states). In the future it could be an error return

Reply:
Problem is that as-pect fails to catch the exception for weird reason... Should I already implement the error return or a Result like return ?

Reply:
no, then let's keep the error slient.
Note that this check shouldn't be useful in the first place: we are supposed to trust what the ABI returns. If it sends something invalid, it's normal to panic